nano/node/node.cpp

```diff
@@ -3280,6 +3281,39 @@ void nano::active_transactions::adjust_difficulty (nano::block_hash const & hash
 	}
 }
 
+void nano::active_transactions::update_active_difficulty ()
+{
+	assert (!mutex.try_lock ());
+	uint64_t difficulty (node.network_params.publish_threshold);
+	auto size_l (roots.size ());
+	// Rough average difficulty estimation
+	switch (size_l)
+	{
+		case 0:
+			break;
+		case 1:
+			difficulty = roots.get<1> ().begin ()->adjusted_difficulty;
+			break;
+		default:
+			difficulty = roots.get<1> ().find ((roots.get<1> ().begin ()->adjusted_difficulty + (--roots.get<1> ().end ())->adjusted_difficulty) / 2)->adjusted_difficulty;
+			break;
+	}
+	difficulty_cb.push_front (difficulty);
+	{
+		uint64_t result (node.network_params.publish_threshold);
```

Comment:
We don't really need this, we can just assign to active_difficulty directly when dividing sum down.

Reply:
cleaned up